README.md

```diff
@@ -50,18 +83,70 @@ Examples can be found on the [examples folder](examples).
 ```bash
 $ cd $GOPATH/src/github.com/libp2p/go-libp2p
 $ make deps
-$ go test ./p2p/<path of module you want to run tests for>
+$ gx test ./p2p/<path of module you want to run tests for>
+```
+
+## Development
+
+### Dependencies
+
+While developing, you need to use [gx to install and link your dependencies](https://github.com/whyrusleeping/gx#dependencies), to do that, run:
+
+```sh
+> make deps
 ```
 
+Before commiting and pushing to Github, make sure to rewind the gx'ify of dependencies. You can do that with:
+
+```sh
+> make publish
+```
+
+### Tests
+
+Running of individual tests is done through `go test <path to test>`
+
+### Packages
+
+List of packages currently in existence for libp2p
 
-## Links
-- [**Specs**](https://github.com/libp2p/specs)
-- [**Website**](https://github.com/libp2p/website)
+| Package            | Version | CI                  |
+|--------------------|---------|---------------------|
+| **Transports**                                     |
+| **Connection Upgrades**                            |
+| **Stream Muxers**                                  |
+| **Discovery**                                      |
+| **Crypto Channels**                                |
+| **Peer Routing**                                   |
+| **Content Routing**                                |
+| **Generics**                                       |
```

Comment:
fair point, specially with all the discussion about generics in go land. Renamed to Miscellaneous, it is the packages that don't fit a specific category.